### PR TITLE
[SIM_FLUTTER-78] [Integrate] Display users dynamically

### PR DESCRIPTION
**Pull request by jstephend-sun**
### Issue links
- [backlog](https://framgiaph.backlog.com/view/SIM_FLUTTER-78)
- [slack](https://sun-philippine.slack.com/archives/C059L7XAXFB/p1687493859253279)

### Definition of Done
- [x] display list of users
- [x] pull down to refresh the users list

### Note
- to see if the pull down refresh is working
<details><summary>Do this temporarily</summary>

file: backend\app\Http\Controllers\UserController.php

image

</details>
- bottom navbar show/hide user list tab if user is admin, will be done in another task

### Recording
- 1

[screen-capture.webm](https://github.com/framgia/sph-flutter/assets/95029803/b6ce7598-5a6b-45b2-9baa-b2c2e13fab7f)

### CODE REVIEW

**Comment by Dbulawan13** [comment]
Comment:

- [ ] Should only display list of users. Admin user should not be included


- [ ] Should not display the current logged in admin user

https://github.com/framgia/sph-flutter/assets/86758474/0c9071b5-6cb8-42df-93d9-fee5d58c6d75

- [ ] Should display all user accounts. 


Comments: 3

**Comment by Dbulawan13** [comment]
<RESOLVED: NOT A BUG>
Comment:

- [ ] Should display all user accounts.
image

Comment: 1

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #27 

Working:
- Able to display users dynamically
- Only Users is displayed. Admin users is not displayed
- Deleted users is not displayed



Testing Hours: 1
Comments: 3
Iteration: 1

File: backend/app/Http/Middleware/LogRequest.php [hunk 1]
```diff
@@ -8,17 +8,17 @@
...
+         */
+        public function handle(Request $request, Closure $next)
+        {
+                $response = $next($request);
+                Log::info('Request Captured', array_merge(['url' => $request->url()], $request->all()));
 
-        return $response;
-    }
+                return $response;
+        }
```
Comment (karlolee50) [inline backend/app/Http/Middleware/LogRequest.php]: this seems to be linters problem? can you verify?

File: backend/app/Models/User.php [hunk 2]
```diff
@@ -102,4 +102,9 @@ public static function searchUsers(String | null $keyword)
             })
             ->get();
     }
+
+    public static function usersList()
+    {
+        return User::where(['is_admin' => false])->get();
+    }
```
Comment (karlolee50) [inline backend/app/Models/User.php]: admin users are not able to see other admin users as well?

File: frontend/lib/src/models/user.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        isAdmin: json["is_admin"],
+        firstName: json["first_name"],
+        lastName: json["last_name"],
+        middleName: json["middle_name"],
+        email: json["email"],
+        address: json["address"],
+        birthday: DateTime.parse(json["birthday"]),
+        deletedAt: json["deleted_at"],
+        createdAt: DateTime.parse(json["created_at"]),
+        updatedAt: DateTime.parse(json["updated_at"]),
```
Comment (karlolee50) [inline frontend/lib/src/models/user.dart]: do we need these data in the FE?

File: frontend/lib/src/models/user.dart [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        "is_admin": isAdmin,
+        "first_name": firstName,
+        "last_name": lastName,
+        "middle_name": middleName,
+        "email": email,
+        "address": address,
+        "birthday": birthday.toIso8601String(),
+        "deleted_at": deletedAt,
+        "created_at": createdAt.toIso8601String(),
+        "updated_at": updatedAt.toIso8601String(),
```
Comment (karlolee50) [inline frontend/lib/src/models/user.dart]: same here

**Review by karlolee50 — DISMISSED** [review]

File: backend/app/Models/User.php (hunk 2 above)
Comment (jstephend-sun) [reply backend/app/Models/User.php]: @karlolee50 that was what was discussed

File: frontend/lib/src/models/user.dart (hunk 3 above)
Comment (jstephend-sun) [reply frontend/lib/src/models/user.dart]: @karlolee50 as of right now, no,
we can remove

File: backend/app/Http/Middleware/LogRequest.php (hunk 1 above)
Comment (jstephend-sun) [reply backend/app/Http/Middleware/LogRequest.php]: i somehow had my spacer on vscode to be 8spaces

**Comment by Dbulawan13** [comment]
Comment:

- [ ] Current logged in admin user should not display in the user list
image

Comment: 1

**Comment by Dbulawan13** [comment]
Comment:

- [ ] Only user accounts were displayed. Admin accounts were not displayed.

https://github.com/framgia/sph-flutter/assets/86758474/12b8554c-cd3c-4ca1-87f6-b3137adeb28e

Comment: 1

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #27 

Working:
- Able to display users and admin accounts dynamically
- Admin accounts display different icon and admin can't delete the admin account
- Deleted users is not displayed




Testing Hours: 2
Comments: 5
Iterations: 3

File: frontend/lib/src/controllers/admin_user_list_controller.dart [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import 'package:frontend/src/models/user.dart';
+import 'package:frontend/src/services/user_service.dart';
+import 'package:get/get.dart';
```
Comment (rdelrosariojr) [inline frontend/lib/src/controllers/admin_user_list_controller.dart]:
```suggestion
import 'package:get/get.dart';

import 'package:frontend/src/models/user.dart';
import 'package:frontend/src/services/user_service.dart';
```

File: frontend/lib/src/features/admin_user_list/admin_user_list_page.dart [hunk 6]
```diff
@@ -1,80 +1,79 @@
 import 'package:flutter/material.dart';
 import 'package:frontend/src/components/breadcrumb.dart';
 import 'package:frontend/src/components/input/search_field.dart';
+import 'package:frontend/src/controllers/admin_user_list_controller.dart';
 import 'package:frontend/src/features/admin_user_list/components/user_list_tile.dart';
+import 'package:frontend/src/models/user.dart';
+import 'package:get/get.dart';
```
Comment (rdelrosariojr) [inline frontend/lib/src/features/admin_user_list/admin_user_list_page.dart]:
```suggestion
import 'package:flutter/material.dart';
import 'package:get/get.dart';

import 'package:frontend/src/components/breadcrumb.dart';
import 'package:frontend/src/components/input/search_field.dart';
import 'package:frontend/src/controllers/admin_user_list_controller.dart';
import 'package:frontend/src/features/admin_user_list/components/user_list_tile.dart';
import 'package:frontend/src/models/user.dart';
```

File: frontend/lib/src/controllers/admin_user_list_controller.dart [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+class AdminUserListController extends GetxController {
+  RxList<User> users = <User>[].obs;
+
+  @override
+  void onInit() async {
+    super.onInit();
+    await getUsers();
```
Comment (RDLegs) [inline frontend/lib/src/controllers/admin_user_list_controller.dart:12]: Is it possible to just call `users.assignAll(await UserService.getUsers())` inside `onInit()` and remove the `getUsers()` method?

**Review by rdelrosariojr — COMMENTED** [review]
reviewed time: 0.5
comments: 2

**Review by RDLegs — APPROVED** [review]

**Review by rdelrosariojr — APPROVED** [review]
reviewed time: 0.5
comments: 2

**Pull request merged**